### PR TITLE
0.1.9

### PR DESCRIPTION
**Pull request by pdyba**

### CODE REVIEW

File: lbz/authz.py [hunk 1]
```diff
@@ -3,18 +3,20 @@
...
 from os import environ
+import warnings
 
 from jose import jwt
 
 from lbz.misc import NestedDict, Singleton
-
 from lbz.exceptions import PermissionDenied, ServerError
 
 # NTH: Consider getting that from SSM
```
Comment (songofroland) [inline lbz/authz.py:17]: Should we remove it now ?

File: tests/test_authz_Authorizer.py [hunk 2]
```diff
@@ -106,23 +111,59 @@ def test_validate_all(self):
         assert self.authz.outcome == ALLOW
 
     def test_validate_fail_all(self):
```
Comment (songofroland) [inline tests/test_authz_Authorizer.py:113]: I would indicate this tests checks expiration or add separate method

File: tests/test_authz_Authorizer.py [hunk 3]
```diff
@@ -106,23 +111,59 @@ def test_validate_all(self):
...
             self.authz.validate("function_name")
         assert self.authz.outcome == DENY
 
     def test_validate_one(self):
         token = self.authz.sign_authz(
-            {"allow": {"res": {"permission_name": {"allow": ALL}}}, "deny": {}}
+            {
+                "allow": {"res": {"permission_name": {"allow": ALL}}},
+                "deny": {},
+                "expiration-at": self.expiration_positive,
```
Comment (songofroland) [inline tests/test_authz_Authorizer.py:127]: not **expires_at** ?

File: tests/test_authz_Authorizer.py [hunk 4]
```diff
@@ -143,7 +188,9 @@ def test_validate_raise_server_error(self):
             self.authz.validate("function_no_name")
```
Comment (songofroland) [inline tests/test_authz_Authorizer.py:188]: Why does it raise server error ? Could you please indicate the cause of that error somewhere?

File: tests/test_authz_Authorizer.py (hunk 2 above)
Comment (pdyba) [reply tests/test_authz_Authorizer.py:113]: this is correct this check does not check exp validation per se.

File: tests/test_authz_Authorizer.py [hunk 5]
```diff
@@ -44,7 +45,11 @@ def setup_class(self):
     def setup_method(self, method):
         self.authz = Authorizer(resource="res")
         self.authz.add_permission("permission_name", "function_name")
-        token = self.authz.sign_authz({"allow": {ALL: ALL}, "deny": {}})
+        self.expiration_positive = (datetime.utcnow() + timedelta(minutes=30)).timestamp()
```
Comment (grzegorzpro) [inline tests/test_authz_Authorizer.py:48]: I'd call it `ts_after_expiration_time` and `ts_before_expiration_time`

File: lbz/authz.py [hunk 6]
```diff
@@ -63,7 +65,16 @@ def set_resource(self, resource_name):
...
-            raise ServerError
+            raise NotAcceptable()
+        if self.expiration is None:
+            warnings.warn(
+                "EXPIRATION_KEY will be mandatory with 0.2 please upgrade Authz provider",
+                DeprecationWarning,
+            )
+        elif self.expiration < datetime.utcnow().timestamp():
+            raise PermissionDenied(
+                f"Your token has expired at {self.expiration} . Please refresh it."
```
Comment (grzegorzpro) [inline lbz/authz.py:76]: Would be nice to convert the timestamp to actual datetime string before displaying, also there is an unnecessary space before the first period

File: lbz/authz.py [hunk 7]
```diff
@@ -63,7 +66,20 @@ def set_resource(self, resource_name):
...
+                "EXPIRATION_KEY will be mandatory with 0.2 please upgrade Authz provider",
+                DeprecationWarning,
+            )
+        if self.iss is None:
+            warnings.warn(
+                "Lack ALLOWED_ISS is a security risk You should add it.",
+                SecurityRiskWarning,
+            )
+        elif self.iss and self.iss != ALLOWED_ISS:
+            print(ALLOWED_ISS)
```
Comment (songofroland) [inline lbz/authz.py:81]: print

File: tests/test_authz_Authorizer.py [hunk 8]
```diff
@@ -131,19 +219,31 @@ def test_validate_one_scope(self):
...
+                "iat": self.iat,
+                "iss": self.iss,
+            }
         )
         self.authz.set_policy(token)
         self.authz.validate("function_name")
         assert self.authz.outcome == LIMITED_ALLOW
         assert self.authz.denied_resource == "self"
 
     def test_validate_raise_server_error(self):
```
Comment (songofroland) [inline tests/test_authz_Authorizer.py:235]: `test_validate_raises_not_acceptable`

**Pull request merged**